src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java

```diff
@@ -416,8 +439,22 @@ private static boolean isSingleLineIf(DetailAST literalIf) {
      * @return true if current lambda statement is single-line statement.
      */
     private static boolean isSingleLineLambda(DetailAST lambda) {
-        final DetailAST block = lambda.getLastChild();
-        return isOnSameLine(lambda, block);
+        final DetailAST lastLambdaToken = getLastLambdaToken(lambda);
+        return isOnSameLine(lambda, lastLambdaToken);
+    }
+
+    /**
+     * Looks for last lambda token.
+     *
+     * @param lambda token to check.
+     * @return last lambda token
```

Comment:
`last lambda token` => `last token in lambda`

Reply:
done